lib/src/configuration.dart

```diff
@@ -3,6 +3,7 @@
 // https://opensource.org/licenses/MIT.
 
 import 'dart:collection';
+import 'package:uuid/uuid.dart';
```

Comment:
I don't think using a full UUID here is necessary. UUIDs are mostly valuable for propagating the notion of "opaque ID" across network or process boundaries. Within a given process, you can call `Object()` and get something that will be `==` to only itself. This will also allow `Configuration.empty()` to remain const.

Reply:
I stayed away from that to be more within an "opaque ID" value which in my mind looked like a string, but that was also because how the current set of constructors are organized.

I reorganized the constructors and which allowed removing the public references to the original config be hidden